src/Microsoft.ML.DnnImageFeaturizer.AlexNet/AlexNetExtension.cs

```diff
@@ -22,7 +22,9 @@ public static class AlexNetExtension
         /// </summary>
         public static EstimatorChain<ColumnCopyingTransformer> AlexNet(this DnnImageModelSelector dnnModelContext, IHostEnvironment env, string outputColumnName, string inputColumnName)
         {
-            return AlexNet(dnnModelContext, env, outputColumnName, inputColumnName, Path.Combine(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "DnnImageModels"));
+            // Substring is  removing initial "file:///" from the codebase string.
+            string executingAssemblyLocation = Directory.GetParent(Assembly.GetExecutingAssembly().CodeBase.Substring(8)).FullName;
```

Comment:
It feels like we should have an internal helper method for this, since we are copying the code in 4 separate places.

Reply:
@eerhardt these 4 separate places are in 4 different projects. So in order to have this in a helper method , the options I can think of
- These all 4 projects have a dependency on Microsoft.ML.OnnxTransform.csproj. We can add a internal helper method in one of those and use internals visible to attribute.
- put the file in one of the projects and use compile item in csproj of other 3 projects. (But we are doing it anywhere in this repo)

which option should I opt for or do u have any other option ?

Reply:
Is there a better method than `.Substring(8)`? I don't know .NET well enough, but most languages/frameworks have a helper function this.

Reply:
We do the first option all over the place.

We are doing the 2nd option here:

https://github.com/dotnet/machinelearning/blob/d1a2962b85138c186d8b4cbbb77c4c458b3abc87/src/Microsoft.ML.CpuMath/Microsoft.ML.CpuMath.csproj#L13-L15

My preference would be for the 2nd option, as it allows for more control (you don't have this internal dependency between projects).

Reply:
// Substring is  removing initial "file:///" from the codebase string.
            string executingAssemblyLocation = 
    Directory.GetParent(Assembly.GetExecutingAssembly().CodeBase.Substring(8)).FullName;

Aside from consolidation into a helper function (good idea), the "file:///" handling seems like it may work for now, but not sure if it'll work in all future versions as well (windows server for example, produces slightly different paths than workstation). If an alternate API can be used instead of looking for the prefix of "file:///", it would seem more stable.